File: RAI/backend/app/models/food_entry.py
from dataclasses import dataclass
from datetime import datetime
from typing import Literal

@dataclass(frozen=True)
class FoodEntry:
    id: str

    user_id: str
    food_id: str

    quantity_g: float

    calories: float
    protein_g: float
    carbs_g: float
    fat_g: float


    consumed_at: datetime
    
    meal_type: Literal[
        "breakfast",
        "lunch",
        "dinner",
        "snack",
        "unspecified",
    ] = "unspecified"

    created_at: datetime | None = None
    updated_at: datetime | None = None

File: RAI/frontend/mobile/types/food-entry.ts
export type MealType = 
    | "breakfast"
    | "lunch"
    | "dinner"
    | "snack"
    | "unspecified"

export type CreateFoodEntryRequest = {
    food_id: string;
    quantity_g: number;
    meal_type: MealType;
    consumed_at: string;
}

export type FoodEntry = {
    id: string;
    food_id: string;
    quantity_g: number;
    calories: number;
    protein_g: number;
    carbs_g: number;
    fat_g: number;
    meal_type: MealType;
    consumed_at: string;
}